mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -106,6 +106,11 @@ class LogisticRegression(override val uid: String)
       case LabeledPoint(label: Double, features: Vector) => (label, features)
     }
     val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    trainOnInstances(instances, handlePersistence)
+  }
+
+  protected[spark] def trainOnInstances(instances: RDD[(Double, Vector)],
```

Comment:
That an option, I figured it would be better to not round trip it through a `DataFrame` since we would have to create a `SQLContext` and the `ml` implementation just rips out the `LabeledPoint`s from the DataFrame as soon as its passed in - but I can do it this way if that would be better :)